api/src/main/java/com/nicknathanjustin/streamercontracts/users/dtos/PublicUser.java

```diff
@@ -1,11 +1,18 @@
 package com.nicknathanjustin.streamercontracts.users.dtos;
 
+import com.nicknathanjustin.streamercontracts.settings.UserSettingsModel;
 import com.nicknathanjustin.streamercontracts.users.externalusers.TwitchUser;
 import lombok.NonNull;
+import lombok.Value;
+import org.springframework.lang.Nullable;
 
+@Value
 public class PublicUser extends User{
 
-    public PublicUser(@NonNull TwitchUser twitchUser) {
+    private final boolean hasPayPalEmail;
+
+    public PublicUser(@NonNull TwitchUser twitchUser, @Nullable UserSettingsModel userSettingsModel) {
```

Comment:
I've read in a couple clean-code books that it should be preferred to have 2 contructors here:

```
public PublicUser(@NonNull final TwitchUser twitchUser) {
```
and
```
public PublicUser(@NonNull final TwitchUser twitchUser, @Nonnull final UserSettingsModel userSettingsModel) {
```

The reasoning is some people think its a code smell to pass 'flags' as input params to methods. They think passing a flag is showing that the method is really doing 2 things and therefore should be 2 methods. In this example the null/not-null settings could be considered a flag.

I agree with that logic and think we should favor 2 contructors here. But its a small thing and i wont block the cr for it.

Reply:
Yeah sure. I don't think we will use the first constructor but it make sense especially since the 2nd parameter can be null.